pkg/controller/service/service_controller_test.go

```diff
@@ -833,45 +833,59 @@ func TestProcessServiceDeletion(t *testing.T) {
 
 }
 
+// Test cases:
```

Comment:
Not sure if it is worth it to put excessive description. I'm hoping the test case description itself would be clear enough. But since you have added it, let's keep it then.

Reply:
There are three factors, finalizer, timestamp and type, means we need 8 testcases. But some cases can be aggregated together. Comment here maybe make sense for readers.